DSCResources/MSFT_xDnsServerADZone/MSFT_xDnsServerADZone.psm1

```diff
@@ -0,0 +1,222 @@
+    param
+    (
+        [Parameter(Mandatory)]
+        [ValidateNotNullOrEmpty()]
+        [System.String]$Name,
+
+        [ValidateSet('None','NonsecureAndSecure','Secure')]
+        [System.String]$DynamicUpdate = 'Secure',
+
+        [Parameter(Mandatory)]
+        [ValidateSet('Custom','Domain','Forest','Legacy')]
+        [System.String]$ReplicationScope,
+
+        [System.String]$DirectoryPartitionName,
+
+        [System.String]$ComputerName,
+
+        [pscredential]$Credential,
+
+        [ValidateSet('Present','Absent')]
+        [System.String]$Ensure = 'Present'
+    )
+    Assert-Module -ModuleName 'DNSServer'
+    Write-Verbose ($LocalizedData.CheckingZoneMessage -f $Name, $Ensure)
+    $cimSessionParams = @{ErrorAction = 'SilentlyContinue'}
+    if ($ComputerName)
+    {
+        $cimSessionParams += @{ComputerName = $ComputerName}
+    }
+    else
+    {
+        $cimSessionParams += @{ComputerName = $env:COMPUTERNAME}
+    }
+    if ($Credential)
+    {
+        $cimSessionParams += @{Credential = $Credential}
+    }
+    $cimSession = New-CimSession @cimSessionParams
+    $getParams = @{
+        Name = $Name
+        CimSession = $cimSession
+        ErrorAction = 'SilentlyContinue'
+    }
+    $dnsServerZone = Get-DnsServerZone @getParams
+    $targetResource = @{
+        Name = $dnsServerZone.ZoneName
+        DynamicUpdate = $dnsServerZone.DynamicUpdate
+        ReplicationScope = $dnsServerZone.ReplicationScope
+        DirectoryPartitionName = $dnsServerZone.DirectoryPartitionName
+        Ensure = if ($dnsServerZone -eq $null) { 'Absent' } else { 'Present' }
+        CimSession = $cimSession
+    }
+    return $targetResource
+} #end function Get-TargetResource
+
+function Test-TargetResource
+{
+    [CmdletBinding()]
+    [OutputType([System.Boolean])]
+    param
+    (
+        [Parameter(Mandatory)]
+        [ValidateNotNullOrEmpty()]
+        [System.String]$Name,
+
+        [ValidateSet('None','NonsecureAndSecure','Secure')]
+        [System.String]$DynamicUpdate = 'Secure',
+
+        [Parameter(Mandatory)]
+        [ValidateSet('Custom','Domain','Forest','Legacy')]
+        [System.String]$ReplicationScope,
+
+        [System.String]$DirectoryPartitionName,
+
+        [System.String]$ComputerName,
+
+        [pscredential]$Credential,
+
+        [ValidateSet('Present','Absent')]
+        [System.String]$Ensure = 'Present'
+    )
+    $targetResource = Get-TargetResource @PSBoundParameters
+    $targetResourceInCompliance = $true
+    if ($Ensure -eq 'Present')
+    {
+        if ($targetResource.Ensure -eq 'Present')
+        {
+            if ($targetResource.DynamicUpdate -ne $DynamicUpdate)
+            {
+                Write-Verbose ($LocalizedData.NotDesiredPropertyMessage -f 'DynamicUpdate', $DynamicUpdate, $targetResource.DynamicUpdate)
+                $targetResourceInCompliance = $false
+            }
+            elseif ($targetResource.ReplicationScope -ne $ReplicationScope)
```

Comment:
Might want to remove the `elseif` and replace with `if`? As it stands, we'll only log one verbose message about a property being incorrect? The Set-TargetResource will still fire.